docs/ABI/Mangling.rst

```diff
@@ -1041,3 +1041,19 @@ Some kinds need arguments, which precede ``Tf``.
 If the first character of the string literal is a digit ``[0-9]`` or an
 underscore ``_``, the identifier for the string literal is prefixed with an
 additional underscore ``_``.
+
+Conventions for constructing names
+----------------------------------
+
+Swift is a language that integrates with multiple languages (C, C++, ObjC,
+ObjC++). To achieve interoperability symbols from other languages have to be
+adapted in order to be integrated into Swift.
+
+Importing C and C++ structs
+~~~~~~~~~~~~~~~~~~~
+
+C and C++ structs use their original names and their owning modules. Since Swift
+metadata symbols and value-witness-tables are not generated by the C compiler,
+Swift generates them when C/C++ module is imported and stores them in the `__C`
+module. This can be observed when mangling a Swift function that accepts a C/C++
+struct as a parameter.
```

Comment:
This would be a great place to show e.g. the symbol name for a C/C++ function and its demangling versus the (de)mangling for its corresponding type descriptor.

Reply:
Done.